internal/firmware/revision.go

```diff
@@ -0,0 +1,121 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package firmware
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type Revision struct {
+	N int
+}
+
+func (r Revision) String() string {
+	if r.N == 0 {
+		return "unset"
+	}
+	if r.N < 0 {
+		return fmt.Sprintf("x%d", -r.N)
+	}
+	return strconv.Itoa(int(r.N))
```

Comment:
Seems like the `int(...)` should be unnecessary here? Though if this is taken verbatim from the snapd project, maybe we want to keep it in sync?

Reply:
I will propagate this change to snapd.